airflow/contrib/kubernetes/pod_launcher.py

```diff
@@ -117,6 +135,35 @@ def read_pod(self, pod):
                 'There was an error reading the kubernetes API: {}'.format(e)
             )
 
+    def _extract_xcom(self, pod):
+        resp = stream(self._client.connect_get_namespaced_pod_exec,
+                      pod.name, pod.namespace,
+                      container=self.kube_req_factory.SIDECAR_CONTAINER_NAME,
+                      command=['/bin/sh'], stdin=True, stdout=True,
+                      stderr=True, tty=False,
+                      _preload_content=False)
+        self.log.info(resp)
+        commands = [
+            'cat {}/return.json'.format(self.kube_req_factory.XCOM_MOUNT_PATH),
+            'kill -s SIGINT 1'
+        ]
+        result = None
+        while resp.is_open():
+            resp.update(timeout=1)
```

Comment:
can some of this logic be separated into its own function(s). Seems like a lot of logic for a single function.